README.md

```diff
@@ -88,12 +82,107 @@ Go to **File > Preferences > File Icon Theme > VSCode Icons**.
+
+But before you can even use them you will have to go to your `settings` and make your magic. The settings will provide you 2 different `array items`:
+
+- `vsicons.associations.files`: [IFileExtension[]](https://github.com/robertohuertasm/vscode-icons/blob/master/src/models/IExtension.ts#L18)
+- `vsicons.associations.folders`: [IFolderExtension[]](https://github.com/robertohuertasm/vscode-icons/blob/master/src/models/IExtension.ts#L23)
+
+Each item of the array represents a file or a folder icon. The functionality is the same for `files` and `folders`.
+
+Note that will important to know what are the currently [supported file extensions / icons](https://github.com/robertohuertasm/vscode-icons/blob/master/src/icon-manifest/supportedExtensions.ts) and [supported folder extensions / icons](https://github.com/robertohuertasm/vscode-icons/blob/master/src/icon-manifest/supportedFolders.ts)
+
+**Some examples**
+```json
+// Adding new extensions to an already supported icon.
+// note: the format must match the existing one. If not, it will use the extension you provide.
+"vsicons.associations.files": [
+  { "icon": "js", "extensions": ["myExt1", "myExt2.custom.js"], "format": "svg" }
+]
+
+// Disabling an already supported icon.
+// note: this is, indeed, the functionality that presets are leveraging.
+"vsicons.associations.files": [
+  { "icon": "js", "extensions": [], "format": "svg", "disable": true }
+]
+
+// Adding a new extension.
+// note: see instructions below on custom icons.
+"vsicons.associations.files": [
+  { "icon": "custom", "extensions": ["custom", "my.custom"], "format": "svg" }
+]
+
+// Changing the icon to an already supported icon.
+"vsicons.associations.files": [
+  { "icon": "newIcon", "extensions": [""], "format": "svg", "extends": "js" }
+]
+
+// Overriding an already supported icon.
+// note: the difference between overrides and extends is that overrides will completely
+// remove the older icon functionality while extends will keep older settings by 
+// putting yours on top.
+"vsicons.associations.files": [
+  { "icon": "myJs", "extensions": ["js", "custom.js"], "format": "svg", "overrides": "js" }
+]
+
+```
+
+**Custom Icons**
+Unfortunately, `VSCode` doesn't support absolute paths when generating the icons css. I'm willing to raise an issue in their repo and submit a PR but in the meantime we've come up with a simple solution.
+
+The idea is that you have to look for your `User folder`:
+
+- Windows: `C:\Users\<your_user>\.vscode\extensions`
+- Linux: `/home/<your_user>/.vscode/extensions`
+- Mac: `/Users/roberto/.vscode/extensions/`
+
+and then create a folder there named like this: `robertohuertasm.vscode-icons.custom-icons`. Then, there you can put all your custom icons but, again, they have to follow `vscode-icons` naming conventions:
+
+- Files: `file_type_<value_of_icon_property>@2x.svg`
```

Comment:
We should add a list with the supported icon file extensions (i.e. `svg`, `png` etc).